src/backend/revoker/scanAttestations.ts

```diff
@@ -22,21 +16,44 @@ export interface AttestationInfo extends Omit<IAttestation, 'revoked'> {
 
 let fromBlock = 0;
 
+/** Extends the `event` with the parameters parsed,
+ *  so that the parameters value extraction is easier and more elegant.
+ *
+ * @param event
+ * @returns the extended event
+ */
+function paramsParser(event: ParsedEvent) {
+  return {
+    ...event,
+    parsedParams: Object.fromEntries(
+      event.params.map((param) => [param.type_name, param.value]),
+    ),
+  };
+}
+
 export async function* scanAttestations() {
   const eventGenerator = subScanEventGenerator(
     'attestation',
     'AttestationCreated',
     fromBlock,
     async (events) => {
-      const claimHashes = events.map(({ params }) => params[1].value);
+      const extendedEvents = events.map((event) => paramsParser(event));
+      const claimHashes = extendedEvents.map(
+        (parsed) => parsed.parsedParams.ClaimHashOf as HexString,
```

Comment:
Would you like some destructuring?
```suggestion
        ({ parsedParams }) => parsedParams.ClaimHashOf as HexString,
```

Reply:
Prettier still leaves it in two lines. So, it is the same for me. Tomato tomato.

Reply:
Changed in: [c4c8371](https://github.com/BTE-Trusted-Entity/socialkyc.io/pull/1957/commits/c4c837189371e8d43a9d0c14de7e4b369dfa19a4)